src/statement.js

```diff
@@ -186,11 +186,13 @@ pp.parseDoStatement = function(node) {
 // is a regular `for` loop.
 
 pp.parseForStatement = function(node) {
```

Comment:
I think I prefer [my approach here](https://github.com/acornjs/acorn/pull/661/files#diff-3d30c156bf8689d04457a16c24b9671eR188), because it gives better positions in parse errors and doesn't require another parameter to `parseFor(In)?`.